sdk/identity/identity/src/credentials/authorizationCodeCredential.ts

```diff
@@ -135,6 +138,12 @@ export class AuthorizationCodeCredential implements TokenCredential {
     scopes: string | string[],
     options?: GetTokenOptions
   ): Promise<AccessToken> {
+    const tenantId = processMultiTenantRequest(
+      this.tenantId,
+      this.allowMultiTenantAuthentication,
+      options
+    )!;
```

Comment:
I see you're using the non-null assertion. When would `processMultiTenantRequest` return `undefined` where that would be ok? We've been bitten by the non-null assertion in other packages and I wonder if it makes more sense to throw an error instead, or at least handle the undefined case.

Reply:
The tenantId has a default for this credential, which is `common`. It’s set on the credential options. processMultiTenantRequest will only pick the getToken options’ tenant if it exists, so either it exists in the options, or is assigned by the user, or is just `common`. Would a comment help?

Reply:
Oh I see. I can throw an error just in case

Reply:
I was hoping we could do some TypeScript magic like this:
```ts
export function processMultiTenantRequest<T extends string | undefined>(
  tenantId: T,
  allowMultiTenantAuthentication?: boolean,
  getTokenOptions?: GetTokenOptions
): T extends string ? string : string | undefined {
  if (
    !allowMultiTenantAuthentication &&
    getTokenOptions?.tenantId &&
    tenantId &&
    getTokenOptions.tenantId !== tenantId
  ) {
    throw new Error(multiTenantErrorMessage);
  }
  if (allowMultiTenantAuthentication && getTokenOptions?.tenantId) {
    return getTokenOptions.tenantId;
  }
  return tenantId as any;
}
```

That works but I had to resort to casting the final return as `any`.

Anyway, it looks like you're really only doing non-null assertion here, I see now your other calls handle undefined. With that said, maybe just add a comment so someone running across it will know `this.tenantId` is always available here.

Reply:
ok I’ll add a comment!

Reply:
Oh look, I can do this!

```ts
    const tenantId = processMultiTenantRequest(
      this.tenantId,
      this.allowMultiTenantAuthentication,
      options
    ) || this.tenantId;
```

Safer ^_^